lib/galaxy/web/framework/webapp.py

```diff
@@ -668,7 +668,8 @@ def get_history( self, create=False ):
         """
         history = None
         if self.galaxy_session:
-            history = self.galaxy_session.current_history
```

Comment:
I see the distinction, toolshed is using the standard galaxywebtransaction, so this isn't as easily overridable as I thought, without some refactoring.